main.rb

```diff
@@ -0,0 +1,213 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = self
+    counter = 0
+    loop do
+      yield arr.to_a[counter]
+      counter += 1
+      break if counter == arr.to_a.length
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    arr = self
+    counter = 0
+    loop do
+      yield(arr.to_a[counter], counter)
+      counter += 1
+      break if counter == arr.to_a.length
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = to_a
+    counter = 0
+    selected_elts = []
+    until counter == arr.length
+      selected_elts << arr[counter] if yield(arr[counter])
+      counter += 1
+    end
+    selected_elts
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each do |elt|
+        return false unless yield(elt) == true
+      end
+      true
+    elsif arg.nil?
+      my_each { |elt| return false if elt.nil? || elt == false }
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each do |elt|
+        return true unless elt.instance_of?(arg.class)
+      end
+      false
+    elsif !arg.nil? && arg.instance_of?(Regexp)
+      my_each { |elt| return false unless elt.match(arg) }
+      true
+    end
+  end
```

Comment:
- [ ] `my_all` is not working as the original `all?` method, also is not passing all test cases. 

### `all?` is an Enumerable that has these characteristics:
- Returns `true` if all elements are not false.
```
[1, 'string', true].all?
# true
```
- Returns `false` if it finds at least one element that is not true
```
[1, 'string', nil].all?
# false
```
- it doesn't change the original array/range/hash
- it can accept as arguments a Class, a RegEx, or a pattern, and returns true or false depending if all the values in the array/range/hash match the condition
- it can use a block but it is not necessary to make the method work

[Enumerable#all?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-all-3F)